tests/src/Unit/SocialPostUserTest.php

```diff
@@ -0,0 +1,457 @@
+  }
+
+  /**
+   * Tests the checkIfUserExists method with account returned.
+   *
+   * @covers Drupal\social_post\User\UserManager::checkIfUserExists
+   */
+  public function testCheckIfUserExistsWithUserReturned() {
+    $this->socialPost->expects($this->any())
+      ->method('getUserId')
+      ->will($this->returnValue(97212));
+
+    $this->userStorage->expects($this->once())
+      ->method('loadByProperties')
+      ->with(['plugin_id' => $this->pluginId, 'provider_user_id' => $this->providerUserId])
+      ->will($this->returnValue([$this->socialPost]));
+
+    $this->userManager->setPluginId($this->pluginId);
+    $this->assertEquals(97212, $this->userManager->checkIfUserExists($this->providerUserId));
+  }
+
+  /**
+   * Tests the getAccountsByUserId method with no account returned.
+   *
+   * @covers Drupal\social_post\User\UserManager::getAccountsByUserId
+   */
+  public function testGetAccountsByUserIdWithNoAccountReturned() {
+    $this->userStorage->expects($this->once())
+      ->method('loadByProperties')
+      ->with(['user_id' => $this->userId, 'plugin_id' => $this->pluginId])
+      ->will($this->returnValue([]));
+
+    $this->userManager->setPluginId($this->pluginId);
+    $this->assertEquals([], $this->userManager->getAccountsByUserId($this->userId));
+  }
+
+  /**
+   * Tests the getAccountsByUserId method with account returned.
+   *
+   * @covers Drupal\social_post\User\UserManager::getAccountsByUserId
+   */
+  public function testGetAccountsByUserIdWithAccountReturned() {
+    $this->userStorage->expects($this->once())
+      ->method('loadByProperties')
+      ->with(['user_id' => $this->userId, 'plugin_id' => $this->pluginId])
+      ->will($this->returnValue([$this->socialPost]));
+
+    $this->userManager->setPluginId($this->pluginId);
+    $this->assertEquals([$this->socialPost], $this->userManager->getAccountsByUserId($this->userId));
+  }
+
+  /**
+   * @covers Drupal\social_post\User\UserManager::getCurrentUser
+   */
+  public function testGetCurrentUser() {
+    $this->currentUser->expects($this->once())
+      ->method('id')
+      ->will($this->returnValue(123));
+
+    $this->assertEquals(123, $this->userManager->getCurrentUser());
+  }
+
+  /**
+   * Tests the addRecord method when user exists.
+   *
+   * @covers Drupal\social_post\User\UserManager::addRecord
+   */
+  public function testAddRecordExist() {
+    $this->socialPost->expects($this->any())
+      ->method('getUserId')
+      ->will($this->returnValue(97212));
+
+    $this->currentUser->expects($this->once())
+      ->method('id')
+      ->will($this->returnValue(456));
+
+    $this->userStorage->expects($this->once())
+      ->method('loadByProperties')
+      ->with(['plugin_id' => $this->pluginId, 'provider_user_id' => $this->providerUserId])
+      ->will($this->returnValue([$this->socialPost]));
+
+    $this->userManager->setPluginId($this->pluginId);
```

Comment:
No need to do all these things, just mock `checkIfUserExists` to return an integer.